### PR TITLE
Streamline dependency gathering

### PR DESCRIPTION
**Pull request by dwlehman**
This is an untested concept of how we can substantially reduce the amount of code we run when gathering dependencies.

### CODE REVIEW

**Comment by richm** [comment]
@dwlehman does this fix https://github.com/linux-system-roles/storage/issues/84 ?

**Comment by dwlehman** [comment]
No, it will not fix #84, but it will fix the broader issue mentioned by @pcahyna, namely that dependency gathering currently "does too much".

**Comment by richm** [comment]
lgtm

**Comment by dwlehman** [comment]
I think this is done now FWIW.

File: tests/tests_deps.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+              - name: test1
+                type: lvm
+                fs_type: xfs
+                state: present
+                mountpoint: '/foo'
+
+    - assert:
+        that:
+                - "{{ blivet_output.packages }} == ['lvm2', 'xfsprogs']"
+                - "{{ not blivet_output.changed }}"
```
Comment (richm) [inline tests/tests_deps.yml:28]: This indentation will likely run afoul of `yamllint`, possibly `ansible-lint` - also below too

Comment (dwlehman) [reply tests/tests_deps.yml]: Good call -- thanks.

**Comment by richm** [comment]
lgtm

**Pull request merged**